app/lib/shared/redis_cache.dart

```diff
@@ -329,6 +329,22 @@ class CachePatterns {
             encode: (OpenIdData v) => v.toJson(),
             decode: (v) => OpenIdData.fromJson(v as Map<String, dynamic>),
           ))[configurationUrl];
+
+  /// When a rate limit is triggered, the cached entry holds the time
+  /// until the block should be applied.
+  Entry<DateTime> rateLimitUntil({
+    required String entryKey,
+    required Duration window,
```

Comment:
Does `window` need to be here? Don't that just make it slower?

I think it's just `rateLimitUntil(String operation, RateLimitScope scope, String value)` where `value` is either a userid or a package name, depending on `scope`.

Reply:
Done. (it can be used in the `set` method as TTL)